examples/z_put.c

```diff
@@ -23,6 +23,9 @@ int main(int argc, char **argv) {
     if (argc > 1) keyexpr = argv[1];
     if (argc > 2) value = argv[2];
 
+    z_owned_bytes_map_t attachements = z_bytes_map_new();
```

Comment:
From a usability point of view, I was thinking about having something like this:

```c
// Options.attachment is z_attachment_t and allocated on the first insert
// I would avoid copying the values on the insert since they are going to be copied anyway for serialization
z_attachment_insert(z_loan(&options.attachment), z_attachment_key("hello"), z_attachment_value(value, strlen(value))); 
```